### PR TITLE
[Android] Always set a non-null Device.Info

### PR DESCRIPTION
**Pull request by alanmcgovern**
When we run under the context of layoutlib the `Context` object
that is created will not implement IDeviceInfoProvider. All this
means is that we will not get change notifications when the
orientation changes, but that's ok! This won't happen anyway.

If we instantiate the Device.Info object then everywhere else in
the code will be able to get access to the screen properties.

Fixes https://bugzilla.xamarin.com/show_bug.cgi?id=44893

### CODE REVIEW

**Comment by alanmcgovern** [comment]
I believe this is a better fix than PR #503 so I'll close that. If this patch is acceptable can we merge it back to the current/next release?

File: Xamarin.Forms.Platform.Android/Forms.cs [hunk 1]
```diff
@@ -190,17 +187,20 @@ static Color GetAccentColor()
...
 		{
-			readonly IDeviceInfoProvider _formsActivity;
+			readonly Context _formsActivity;
 			readonly Size _pixelScreenSize;
 			readonly double _scalingFactor;
 
 			Orientation _previousOrientation = Orientation.Undefined;
 
-			public AndroidDeviceInfo(IDeviceInfoProvider formsActivity)
+			public AndroidDeviceInfo(Context formsActivity)
```
Comment (adrianknight89) [inline Xamarin.Forms.Platform.Android/Forms.cs]: What happens if you remove the parameter and just use `Context` everywhere inside this class? I'm not sure what kind of purpose `IDeviceInfoProvider` serves.

Comment (alanmcgovern) [inline Xamarin.Forms.Platform.Android/Forms.cs]: It has an extra event which is not available on a pure `Context` object

Comment (adrianknight89) [inline Xamarin.Forms.Platform.Android/Forms.cs]: What if you cast `Context` to an `Activity`? Shouldn't you see the event then?

Comment (alanmcgovern) [inline Xamarin.Forms.Platform.Android/Forms.cs]: There's no guarantee it is an `Activity`. The information we need exists on `Context` and we are guaranteed to have a `Context`, so we should take that information.

If we are lucky and have an `Activity` then we can definitely cast to that and get more info, or if it's an IDeviceInfoProvider we could cast to that. However we cannot rely on that always being true.

Comment (adrianknight89) [inline Xamarin.Forms.Platform.Android/Forms.cs]: If there is no guarantee that it's an `Activity`, then `_formsActivity` should probably be renamed to `context`. Right now, it implies that it's an `Activity`.

Comment (hartez) [inline Xamarin.Forms.Platform.Android/Forms.cs]: I agree with @adrianknight89 on the naming; `_formsActivity` should be renamed `_context`. The bad naming is a pre-existing condition here, so I'm not going to reject the PR if it doesn't get changed, but it'd be nice.

File: Xamarin.Forms.Platform.Android/Forms.cs [hunk 2]
```diff
@@ -224,7 +224,8 @@ public override double ScalingFactor
 
 			protected override void Dispose(bool disposing)
 			{
-				_formsActivity.ConfigurationChanged -= ConfigurationChanged;
+				if (_formsActivity is IDeviceInfoProvider)
```
Comment (hartez) [inline Xamarin.Forms.Platform.Android/Forms.cs]: This `if` block needs to be surrounded by `if (disposing){}` so it doesn't run in the destructor.

File: Xamarin.Forms.Platform.Android/Forms.cs [hunk 3]
```diff
@@ -224,7 +224,8 @@ public override double ScalingFactor
 
 			protected override void Dispose(bool disposing)
 			{
```
Comment (hartez) [inline Xamarin.Forms.Platform.Android/Forms.cs]: This needs to properly handle double disposal (i.e., it needs a 'disposed' flag). For an example, see https://github.com/xamarin/Xamarin.Forms/blob/ba6caad23ed7abdb1661cc61114c24eb3ae09da1/Xamarin.Forms.Platform.Android/AppCompat/MasterDetailContainer.cs#L113

**Pull request merged**